service/collector/trusted_advisor.go

```diff
@@ -88,6 +88,10 @@ func init() {
 }
 
 func (c *Collector) collectAccountsTrustedAdvisorChecks(ch chan<- prometheus.Metric, clients []aws.Clients) {
+	if !c.trustedAdvisorSupported {
+		return
+	}
```

Comment:
Isn't this a bit too late? Can't we prevent the collector to be initiated in the first place when there is even a separate collector for it?

Reply:
That was my first thought too, but then I noticed that it was being done in `func init() {}`. I though this would be fine, but I'll check for a change so that it wouldn't even get initialized.

Reply:
Check the `azure-operator` and how the collector is initialized there. I would like to use such an approach.